WordPress/gutenberg
# Add a new public method for getting the selector for a block

## What?
This PR came about from the conversation that was had [here](https://github.com/WordPress/gutenberg/pull/45089#discussion_r1011954379) between @oandregal, @alecgeatches  and I about having a public method to get the selector for a block instead of exposing the `get_blocks_metadata function.

## Why?
Given the [recent drive to do a code quality check for theme.json APIs](https://github.com/WordPress/gutenberg/issues/45171), it was mentioned to not expose the `get_blocks_metadata` function like we proposed in [this PR](https://github.com/WordPress/gutenberg/pull/45089). Instead, it's better to make a simpler public method that when given the block name would return the selector for that block.

This is using existing cached methods to do its job so the lookup time is minimal, and it ensure it only returns the basic information needed which is the selectors.

## How?
It's using a simpler version of the `get_blocks_metadata` function by:

- getting the registered blocks
- looking up the name of block based on the name provided
- checking to see what selectors are available, and returning one accordingly

## Testing Instructions
Make a test plugin that essentially calls the following:

```$looked_up_selector = wp_theme_get_css_selector_for_block( $block_name );```

for a block. Ensure the right value is given back.
